runtime/activations/activations.go

```diff
@@ -158,13 +164,13 @@ func (a *Activations) Pop() {
 // CurrentOrNew returns the current activation,
 // or if it does not exists, a new activation
 //
-func (a *Activations) CurrentOrNew() Activation {
-	current := a.current()
+func (a *Activations) CurrentOrNew() *Activation {
+	current := a.Current()
 	if current == nil {
-		return NewActivation()
+		return NewActivation(nil)
```

Comment:
It's surprising to me that if there is no current activation, that you don't create one _and make it the current_. Clearly this logic hasn't changed with this PR, but maybe a comment is warranted here?